tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/RunConfigurationSelector.java

```diff
@@ -264,16 +265,39 @@ private void uncheckAllMutators() {
      */
     private void toggleMutator(SWTBotTable table, Mutators mutator) {
         table.getTableItem(mutator.getDescriptor()).toggleCheck();
+        pressApplyIfEnabled();
+    }
+
+    private void pressApplyIfEnabled() {
+        SWTBotButton apply = bot.shell(RUN_CONFIGURATIONS).bot().button("Apply");
+        if (apply.isEnabled()) {
+            apply.click();
+        }
+    }
+
+    /**
+     * @return the run configuration shell or null, if it is not open
+     */
+    private SWTBotShell getRunConfigShellIfOpen() {
+        // look if shell is already open
+        for (SWTBotShell shell : bot.shells()) {
+            if (shell.getText().equals(RUN_CONFIGURATIONS)) {
+                shell.activate();
+                return shell;
+            }
+        }
+        return null;
     }
 
     /**
      * Closes the configuration shell and applies, if possible.
      */
-    private void closeConfigurationShell() {
-        SWTBotShell shell = bot.shell(RUN_CONFIGURATIONS);
-        SWTBotButton apply = shell.bot().button("Apply");
-        if (apply.isEnabled()) {
-            apply.click();
+    @Override
+    public void close() {
+        SWTBotShell shell = getRunConfigShellIfOpen();
+        if (shell == null) {
+            // shell is not open
+            return;
         }
         shell.bot().button("Close").click();
```

Comment:
Since you're working on this, maybe it's the right moment to also wait for the shell to be closed with the corresponding SWTBot condition?

Reply:
Is now implemented